fastlane_core/lib/fastlane_core/ui/implementations/shell.rb

```diff
@@ -150,6 +150,19 @@ def password(message)
 
     private
 
+    def encode_as_utf_8_if_possible(message)
+      orig_encoding = message.encoding
+
+      # Try some common encodings and return the first with valid codings
+      ['UTF-8', 'UTF-16', 'ISO-8859-1'].each do |dest_encoding|
+        message.force_encoding(dest_encoding)
+        return message.encode('UTF-8', dest_encoding) if message.valid_encoding?
+      end
+
+      message.force_encoding(orig_encoding)
+      return message
+    end
```

Comment:
And also as a piece of Ruby source code, this is not clear to me. I might want to improve this part.

This makes it a bit more Ruby-ish  code with using constant values and Object#dup not to mutate given parameter. Please document the use cases of encodings in the comment 🙇 

```suggestion
    def encode_as_utf_8_if_possible(message)
      return message if message.valid_encoding?
      
      # Try following common encodings and return the first with valid codings
      # 
      # * UTF-16 - used by genstrings
      # * ISO8859-1 - used by blah blah 
      test_string = message.dup
      [Encoding::UTF_16, Encoding::ISO8859_1].each do |dest_encoding|
        return message.encode(Encoding::UTF_8, dest_encoding) if test_string.force_encoding(dest_encoding).valid_encoding?
      end

      # return the original message if no valid encoding is found
      message
    end
```